Sam’s summer research plan
================

  - [Overview](#overview)
  - [Goals](#goals)
      - [Learning goals](#learning-goals)
      - [Output goals](#output-goals)
  - [Expectations](#expectations)
      - [Coby’s expectations](#cobys-expectations)
      - [Sam’s expectations](#sams-expectations)
  - [Timeline](#timeline)

**last updated:** 2021-05-10

-----

## Overview

This repository will (hopefully) serve as a central location for Sam’s
projects this summer. We’ll set up some mutual expectations, goals, and
a timeline here, and amend as necessary.

-----

## Goals

### Learning goals

#### Bioinformatiics

1.  Rstudio + RMarkdwon
2.  Version control: git and github integration
3.  Basic shell programming
4.  Basic cluster computing
5.  \+/- Snakemake

#### Lab techniques

1.  Cell culture
2.  Electronic lab notebook documentation

### Output goals

1.  Collaborate on RNAseq analysis of exogenous/endogenous FeLV dataset,
    including:
      - Troubleshooting analytical pipeline
      - Generating and writing up preliminary results
      - Generating associated figures
2.  Fully proficient in (and able to teach me\!) mammalian fibroblast
    cell culture protocols.
      - Obtain training from Raegan before she leaves in July.
      - Additional training as necessary from Laura, other lab mates.
3.  Help develop project design for larger FeLV cell culture + genome
    sequencing project.

-----

## Expectations

### Coby’s expectations

Much of what you’ll be doing is computer-based and independent. This is
great, but it can also be daunting and/or hard to keep to a schedule and
self-motivate. To help with this, let’s set up the following
expectations:

1.  Weekly in-person (or virtual) meetings to go over progress.
2.  Email communication as necessary for troubleshooting.
      - Since we’re all working independently on a bunch of different
        projects, it can be hard to keep track of what’s going on. I’ll
        check in with you minimally weekly, and more frequently if you
        prefer. If you need help with anything, please feel free to ask
        at any point\!
3.  Time commitment: 40 hrs/week.
4.  That you learn some useful tools that will be beneficial to you for
    future projects and career goals.
5.  That you have fun\!

### Sam’s expectations

\<insert your expectations here once you figure out git/github 👍 \>

## Timeline

| Due date | Goal                                                                                                                                                        | Date completed | Output notes/links |
| -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- | ------------------ |
| 5/14/21  | Meet with Raegan to go over summer cell culturing plan                                                                                                      |                |                    |
| 5/14/21  | [Classes 1-5 of online data science class](%22https://nt246.github.io/NTRES6940-data-science/syllabus.html%22)                                              |                |                    |
| 5/14/21  | Read background literature on FeLV, generate annotated bibliography                                                                                         |                |                    |
| 5/14/21  | Research+choose best electronic lab notebook, whether github integration is possible                                                                        |                |                    |
| 5/21/21  | Start cell culture with Raegan                                                                                                                              |                |                    |
| 5/21/21  | Obtain Summit account                                                                                                                                       |                |                    |
| 5/21/21  | [Chapters 1-4 of Eric Anderson’s Bioinformatics Handbook](%22https://eriqande.github.io/eca-bioinf-handbook/essential-unixlinux-terminal-knowledge.html%22) |                |                    |
| 5/21/21  | [Classes 6-9 of online data science class](%22https://nt246.github.io/NTRES6940-data-science/syllabus.html%22)                                              |                |                    |
