###

#### edgR: differential expression tests ####

# Dataset: endogenous/exogenous FeLV in vitro gene expression (fibroblasts infected and uninfected, PMBCs uninfected only)

# This code will do the following:
# 1. Generate tissue-specific filtered and normalized DGElists
# 2. Write tables of log-transformed, filtered counts
# 3. Generate a tissue-specific PCA
# 4. Design GLMs, estimate dispersion, and fit models
# 5. Generate DGE contrasts to be made
# 6. Identify sig DEGs in each comparison based on likelihood ratio tests (or quasi-likelihood F-tests which are more conservative), order genes by logFC and FDR<0.05
# 7. Write output tables of DEGs

# Preconditions: tissue-specific gene-level counts data, tissue-specific sample metadata

###


library(edgeR)
library(tidyverse)
library(matrixStats)
library(pheatmap)
library(here)


#start with tissue-specific counts and metadata, with outliers removed: files previously generated by DGE_dataexplore.R
#load data for each tissue separately and run code

##counts and metadata
dat <- read_delim("results/edgeR/counts_all.txt", delim="\t")
metadat <- read_delim("results/edgeR/meta_all.txt", delim="\t")

#remove puma samples
# counts <- dat %>% select(!contains("Mischief")) 
# meta <- metadat %>% filter(cat_id!="Mischief")

#uninfected/baseline only: fibroblasts vs. PMBCs
# counts <- dat %>% select(!contains(c("Mischief", "PLUS")))
# meta <- metadat %>% filter(cat_id!="Mischief") %>% filter(status=="uninfected")

#uninfected vs infected fibroblasts
counts <- dat %>% select(!contains(c("Mischief", "44", "45")))
meta <- metadat %>% filter(cat_id!="Mischief") %>% filter(cell_type=="fibroblast")

# group <- meta$cell_type
group <- meta$status

#### 1. Create DGElist, filter, and calcNormFactors ####
dat <- DGEList(counts=counts[,-(1)], group=group, genes=counts[,1])

#head(dat$samples$group)
keep_counts<-rowSums(cpm(counts[,-(1:5)])>1) >= 0.25 * ncol(dat) #filter >1 cpm in >= 1/4 of samples
dat.filt<-dat[which(keep_counts==T), , keep.lib.sizes=FALSE]
dim(dat.filt)
dat.norm<-calcNormFactors(dat.filt)


#### 2. Log-transform, and write tables for downstream analyses ####
#log-transform, all counts
dat.logCPM <- cpm(dat.norm, log=TRUE, prior.count = 1, normalized.lib.sizes = TRUE) 

# write table for downstream (e.g. WGCNA)
# dat.logCPM.tbl <- cbind(dat.filt$genes, dat.logCPM)
# write_delim(dat.logCPM.tbl, "bp05_edgeR/counts_liv_filt0.66_log.txt", delim="\t") 


#### 3. Generate PCA ####
# Cell_type =c("#008080","#ef6079")
# Infection_status = c("#00b6bd", "lightgrey"),


# PCA:
pca<-prcomp(t(dat.logCPM))
s<-summary(pca)
s$importance
scores<-as.data.frame(pca$x)
# PCA by Treatment
pca <- ggplot(data=scores, aes(x=PC1, y=PC2, group=group)) + 
  geom_point(aes(color=group)) +
  scale_color_manual(values=c("#00b6bd", "lightgrey")) +
  # scale_color_manual(values=c("#008080", "#ef6079")) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1)) +
  theme(panel.background = element_blank(), panel.grid = element_blank()) +
  xlab ("PC1 (84.8%)") +
  ylab ("PC2 (4.9%)") +
  guides(colour = guide_legend(override.aes = list(size=2), title="FeLV status")) 
pca + ggrepel::geom_text_repel(aes(label=meta$cat_id))
# ggsave(file="plots/pca_felv_status.jpg")

#### 4. Design model, estimate dispersion, fit model ####
# design matrix
design.mat <- model.matrix(~ 0 + dat.norm$samples$group)

colnames(design.mat) <- levels(dat.norm$samples$group)
design.mat

#estimate dispersion
d1 <- estimateGLMCommonDisp(dat.norm, design.mat, verbose = TRUE)
d1 <- estimateGLMTrendedDisp(d1,design.mat, method="power")
d1 <- estimateGLMTagwiseDisp(d1,design.mat)
# plotBCV(d1)

# fit model
#quasi-likelihood model (more conservative type I error rate control by reflecting uncertainty in estimating the dispersion for each gene)
#fit.qlf <- glmQLFit(d1, design.mat, robust=TRUE)
#plotQLDisp(fit.qlf)

#GLM model
fit.glm <- glmFit(d1, design.mat)


#### 5. Set up contrasts ####
my.contrasts <- makeContrasts(
  inf_uninf = uninfected - infected,
  # celltype = PBMC - fibroblast,
  levels=design.mat
)


#### 6. Run LR/QLF tests on all contrasts ####

## Individually:
test <- glmLRT(fit.glm, contrast=my.contrasts[,"inf_uninf"])
topDE <- topTags(test, adjust.method="BH", n=NULL)
summary(decideTests(test, p.value = 0.05))

## as loop:
contrast_list <- as.list(colnames(my.contrasts))

topDE_sig <- list()

for (f in 1:length(contrast_list))
{
  #test <- glmQLFTest(fit.qlf, contrast=my.contrasts[,f])
  test <- glmLRT(fit.glm, contrast=my.contrasts[,f])
  topDE <- topTags(test, adjust.method="BH", n=NULL)
  topDE_order <- topDE[order(topDE$table$logFC),]
  topDE_sig[[f]] <- topDE_order[topDE_order$table$FDR<0.05,]
}
names(topDE_sig) <- unlist(contrast_list)

view(topDE_sig$inf_uninf$table)


#### 7. Write tables ####
#change prefix to match tissue
# setwd("results/edgeR")
# 
# lapply(1:length(topDE_sig),
#        function(f) write.table(topDE_sig[[f]], file = paste0("sigDGE_",names(topDE_sig[f]),".txt"), row.names=F, quote=F, sep="\t"))

